feat: latest MCM changes

kirgene opened the pull request.

kalinkrustev committed:
feat: add topology spread constraints for redis leader and follower

kalinkrustev committed:
fix: update cloud platform and k8s cluster type to use local variables

kalinkrustev committed:
feat: add init container for network readiness in microk8s patch job

kalinkrustev committed:
fix: ensure imagePullPolicy is set for init-microk8s container in patch job

kalinkrustev committed:
fix: move calico patch to iac-ansible-collection-roles

kalinkrustev committed:
fix: avoid blocked status

kalinkrustev committed:
Merge pull request #586 from mojaloop/fix/calico

improve Redis topologySpreadConstraints to be zone aware

kalinkrustev committed:
fix: set the iac and ansible tags in submoduleRevisions

kalinkrustev committed:
Merge pull request #592 from mojaloop/fix/submodule-revisions

fix: set the iac and ansible tags in submoduleRevisions

muzammil360 committed:
Optimize mimir ingester memory requirement + drop env prometheus labels and metrics to save on resources (#585)

move prometheus-stats dashboard to lac-modules
restrict mimir block size to 1hr only
add prometheus exporter enabling flag
drop env prometheus labels and metrics to save on resources
fix env alerts

muzammil360 committed:
remove unused timeseries and labels from cc prometheus  (#615)

- Remove unused time series and labels from cc prometheus
- Added network-cost-estimation dashboard to tac-modules
- moved prometheus-remote-write dashboard to iac-modules
- Fix netbird-management dashboard to use __rate_interval
- Do not drop id label in kubelet job to avoid errors in mimir
- Set loki replication factor to 1 by default
- Added network quick view dashboard

muzammil360 committed:
Use Istio dashboards with dynamic rate (#643)

kirgene committed:
add new mcm version

kirgene committed:
remove duplicates

kirgene committed:
fix indents

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
add mcm rbac perms

kirgene committed:
add mcm rbac perms

kirgene committed:
add mcm rbac perms

kirgene committed:
add mcm rbac perms

kirgene committed:
add mcm rbac perms

kirgene committed:
wip

kirgene committed:
rbac changes

kirgene committed:
implement keto roles injection into the kratos user session traits

kirgene committed:
wip

kirgene committed:
separate kratos-role-webhook.js

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
move dfsp realm into ory realm

kirgene committed:
feat: refactor variables and secrets

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
feat: update default-config

kirgene committed:
wip

kirgene committed:
feat: replace mcm_oidc_secret

kirgene committed:
chore: trigger

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
debug

kirgene committed:
debug

kirgene committed:
debug

kirgene committed:
revert smtp changes

kirgene committed:
revert smtp changes

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
move resource

kirgene committed:
shorter secret names

kirgene committed:
secret optimize

kirgene committed:
fix vaultsecret

kirgene committed:
refactor secrets

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
fix: rbac rules

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kleyow committed:
chore: remove reporting-events-processor-svc override (#690)

kirgene committed:
wip

kirgene committed:
add mailhog

kirgene committed:
Merge remote-tracking branch 'origin/main' into feat/mcm

kirgene committed:
add missing file

kirgene committed:
fix mailhog app

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
redirect on auth error

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
version bump

kirgene committed:
chore: fix

kirgene committed:
Merge pull request #707 from mojaloop/feat/mcm

feat: implement fine-grained RBAC system for MCM with Ory Keto integration

kirgene committed:
Merge remote-tracking branch 'origin/main' into feature/storage-cluster

kirgene committed:
fix var

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

KhaledSaiidi committed:
bulk_enabled in mcm tf

kirgene committed:
reverted changes from main

kalinkrustev commented on terraform/gitlab/ci-templates/k8s-cluster/.gitlab/scripts/dictmerge.py:
```diff
@@ -81,10 +81,14 @@ def mergeListOfDicts(data1, data2, fileName, outputFilename, fileType):
+                data1 = json.load(f)
+        except json.JSONDecodeError:
+            print("  Could not parse the default config file", default_config_file," assigning empty dict")
+            data1 = {}
```
As the default comes from IaC, it is best to crash if it cannot be parsed. Proceeding with an empty config can be dangerous or lead to other errors, which will be hard to trace to the silent failure here.

kalinkrustev commented on terraform/gitops/generate-files/templates/istio/istio-main/values-istio-istiod.yaml.tpl:
```diff
@@ -181,6 +181,7 @@ meshConfig:
         statusOnError: "500"
         pathPrefix: /decisions
         includeRequestHeadersInCheck: ["authorization", "cookie"]
+        headersToUpstreamOnAllow: ["x-user", "x-email", "x-roles", "x-client", "x-dfsp-id"]
```
DRPP: We need to add automated test that ensures these headers are properly handled if they exist in the outside request.

kalinkrustev commented on terraform/gitops/generate-files/templates/mcm/kustomization.yaml.tpl:
```diff
@@ -1,10 +1,9 @@
   - vault-certificate.yaml
   - vault-agent.yaml
-  - vault-secret.yaml
-  - keycloak-realm-cr.yaml
```
DRPP: Not sure if deleting the DFSP realm will work. This likely needs a full test starting from cluster creation.

kalinkrustev commented on terraform/gitops/generate-files/templates/keycloak/post-config/keycloak-ingress.yaml.tpl:
```diff
@@ -42,7 +42,7 @@ spec:
       http:
         paths:
-          - path: /realms/${keycloak_dfsp_realm_name}/protocol/openid-connect
+          - path: /realms/${keycloak_hubop_realm_name}/protocol/openid-connect
```
Cross check if deleting the DFSP realm can work.

kalinkrustev commented on terraform/gitops/generate-files/templates/mcm/rbac.yaml.tpl:
DRPP: to ensure this works, we need automated test for all the roles.

kirgene committed:
wip

kalinkrustev commented:
@kirgene thanks for preparing this. We are keen to include it. I am going to leave some comments. The ones prefixed with `DRPP:` are meant to remind us what we need to test when including this change.

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kalinkrustev commented on terraform/gitops/generate-files/templates/mcm/values-mcm.yaml.tpl:
```diff
@@ -1,114 +1,118 @@
+    enabled: true
+    hubOidcProviderUrl: "https://${keycloak_fqdn}/realms/${keycloak_hubop_realm_name}/protocol/openid-connect"
+  image:
+    version: sha-cd633d8e7d10b0326cf907a6ae4e8906caa1cb17
```
Unclear which version this is

kalinkrustev commented on terraform/gitops/generate-files/templates/ory/keto-batch-auth.js:
We should avoid implementing APIs in the IaC, as it lacks automated testing, coverage, etc.

kalinkrustev commented on terraform/gitops/generate-files/templates/ory/kratos-role-webhook.js:
We should avoid implementing APIs in the IaC, as it lacks automated testing, coverage, etc.

kalinkrustev commented on terraform/gitops/generate-files/templates/pm4ml/keycloak-realm-cr.yaml.tpl:
```diff
@@ -14,7 +14,7 @@ spec:
     revokeRefreshToken: false
     refreshTokenMaxReuse: 0
-    accessTokenLifespan: 300
+    accessTokenLifespan: 3600
```
Unsure if this is fit for production

kalinkrustev commented on terraform/gitops/stateful-resources/templates/stateful-resources/strimzi/kafka/kafka-with-dual-role-nodes.yaml.tpl:
DRPP: Kafka upgrade needs to be carefully tested

kalinkrustev commented on terraform/gitops/stateful-resources/templates/stateful-resources/values-redis.yaml.tpl:
```diff
@@ -1574,7 +1574,7 @@ volumePermissions:
     registry: docker.io
     repository: bitnami/os-shell
-    tag: 12-debian-12-r47
+    tag: latest
```
Latest tags are not good for ensuring stable operation

kalinkrustev commented on terraform/k8s/default-config/mcm-values-override.yaml:
```diff
@@ -1,6 +1,7 @@
 ui:
   image:
-    version: 1.8.4
+    version: sha-cd633d8e7d10b0326cf907a6ae4e8906caa1cb17
```
unclear version

kalinkrustev commented on terraform/k8s/default-config/mojaloop-rbac-permissions.yaml:
```diff
@@ -365,10 +365,140 @@ roles:
   - auditSettlement
   - report-settlement-audit
+
+# MCM Roles
```
Not sure what `mta` and `pta` represent

kalinkrustev commented on terraform/k8s/default-config/mojaloop-stateful-resources-local-operator.yaml:
DRPP: version changes here require full testing

kalinkrustev commented on terraform/k8s/default-config/mojaloop-vars.yaml:
```diff
@@ -112,3 +112,12 @@ workload_definitions:
+smtp_auth: false
+smtp_from: "mailer@sw005.hub005.mojaloop.io"
+smtp_from_display_name: "Mojaloop Hub"
+smtp_reply_to: "noreply@sw005.hub005.mojaloop.io"
```
unsure if it is good to have such defaults

kalinkrustev commented on terraform/k8s/default-config/stateful-resources-operators.yaml:
DRPP: version changes here require full testing

kirgene commented:
@kalinkrustev thanks for the comments, these changes come from the main branch and have already been fully tested. They’ve been around for a while (several of months), so they’re not new changes.

kalinkrustev commented:
> @kalinkrustev thanks for the comments, these changes come from the main branch and have already been fully tested. They’ve been around for a while (several of months), so they’re not new changes.

The problem is that they were not tested with the storage cluster branch, which is used for DRPP. This is the branch where active development happened for quite some time and includes a lot of important bug fixes and improved functionality.

kalinkrustev committed:
feat: add cloud platform variable to Mojaloop module for conditional template rendering

kirgene committed:
chore: replace MailHog with Mailpit for email testing

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
Merge branch 'feature/storage-cluster' into feat/mcm-for-storage-cluster

kirgene committed:
wip

kirgene committed:
wip

kirgene committed:
fix: escape the $ character in the template, so ArgoCD passes through ${HOSTNAME##*-} literally

kirgene committed:
fix: add missing creds

kirgene committed:
fix: VaultBackupConfig sync-wave ordering

VaultBackupConfig needs AWS creds from ExternalSecrets (wave 10) but was
running at wave 0, causing a deadlock. Moving to wave 15.

kirgene committed:
fix: output private_dns_zone_id for aws/base-k8s

The private_dns_zone_id output was hardcoded to empty string,
causing dns-utils app to fail validation with "Required value" error.

kirgene committed:
fix: derive backup bucket region from cloud_region by default

kirgene committed:
fix: derive vault backup bucket region from cloud_region by default

kirgene committed:
fix: add sc-kubernetes-provider for non-private-cloud deployments

kirgene committed:
fix: add sc-kubernetes-provider for AWS deployments

kirgene committed:
fix: add percona-operator to CC deployment

kirgene committed:
revert: remove sc-kubernetes-provider changes (using RDS instead)

kirgene committed:
fix: use backup_full_retention env var in RDS config

kirgene committed:
fix: ArgoCD Application health check Lua string.match not available in sandbox

kirgene committed:
fix: HCL syntax in RDS password policy

kirgene committed:
Merge branch 'feature/storage-cluster' into feat/mcm-for-storage-cluster

kirgene committed:
fix: remove unused istio_egress_gateway vars

kirgene committed:
fix: add missing mailpit_namespace variable

kirgene committed:
fix: pre-install community.docker from GitHub to bypass Galaxy 500 error

kirgene committed:
revert: remove Galaxy workaround

kirgene committed:
fix(ci): fetch tags in cached repos before init

kirgene committed:
fix(external-dns): remove wrapper key for kustomize helmCharts compatibility

kirgene committed:
Add MCM RBAC validation test job template

Adds a Kubernetes Job that runs MCM RBAC tests using TTK to validate
portal admin and DFSP operator permissions after MCM deployment.

kirgene committed:
feat: use ml-ttk-test-mcm-rbac helm chart for MCM RBAC tests

- Remove standalone mcm-rbac-test-job.yaml.tpl
- Add ml-ttk-test-mcm-rbac config to values-mojaloop.yaml.tpl
- Update mojaloop_helm_version to 17.1.0-22864-2b5d44b
- Add ttk_mcm_rbac_tests_enabled, mcm_namespace, mailpit_namespace vars

kirgene committed:
feat: use ory-services helm chart for keto-batch-auth and kratos-role-webhook

- Add ory-services chart config, remove inline JS scripts
- Update service refs to ory-services-* prefix
- Add ml_ory_services_chart_version and ml_ory_services_image_version vars

kirgene committed:
fix: add empty placeholder tpl files for terraform state cleanup

kirgene committed:
fix: remove placeholder tpl files after state cleanup

kirgene committed:
fix: use ory-services chart dev version 0.1.0-22864-2b5d44b

kirgene committed:
fix: use mojaloop_helm_repo for ory-services chart

kirgene committed:
feat(keycloak): enable HTTP for ambient mesh compatibility

kirgene committed:
feat(keycloak): add waypoint and ingress-use-waypoint label for ambient mode

kirgene committed:
fix(keycloak): pass namespace_meta to common_stateful_resources module

kirgene committed:
fix(istio): enable autoMtls for ambient mode compatibility

kirgene committed:
feat(keycloak): separate hub-operators and dfsps realms with configurable display names

- Add dfsps realm for MCM-managed DFSP users
- Configure Kratos OIDC providers for both realms
- Add display_name variable for realms (used in Kratos label and Keycloak displayName)
- Update Keycloak operator version to 26.5.2

kirgene committed:
feat(mcm): migrate connection-manager to mojaloop/vnext subcharts

- Remove standalone MCM helm chart deployment (values-mcm.yaml.tpl, values-mcm-override.yaml.tpl)
- Add connection-manager as subchart in values-mojaloop.yaml.tpl and values-vnext.yaml.tpl
- Remove helmCharts section from MCM kustomization (MCM supporting resources still deployed via kustomization)
- Clean up old MCM chart references (mcm_chart_version, mcm_values_override_file)
- Simplify variables by reusing existing ones instead of mcm_ prefixed duplicates
- Add DFSP realm support with separate Keycloak realm for DFSP/MCM managed resources

kirgene committed:
fix: remove duplicate variable declarations from mcm.tf

kirgene committed:
fix: add missing hubop_oidc variable declarations

Add variable declarations for hubop_oidc_client_id and
hubop_oidc_client_secret_secret that were missing in mojaloop module.

kirgene committed:
fix: add missing vnext variable declarations
